api/envoy/config/listener/v3/listener.proto

```diff
@@ -339,6 +339,17 @@ message Listener {
   // provided net.core.somaxconn will be used on Linux and 128 otherwise.
   google.protobuf.UInt32Value tcp_backlog_size = 24;
 
+  // The maximum number of connections to accept from the kernel per socket
```

Comment:
It's a property of each listener; it shouldn't be global, and it should be included in LDS updates, so I think it either needs to live here or in a sub-message of the listener.

Reply:
> It's a property of each listener; it shouldn't be global, and it should be included in LDS updates, so I think it either needs to live here or in a sub-message of the listener.

I agree that it should be per-listener.
However, IMHO the field seems to be too low-level (implementation dependent). A different client may have an implementation that uses something other than event-loop iterations. This implies that the field may not need to be part of the API.
I'm not aware of anywhere else that this should be added (other than here), but just want to make that I'm not missing something.

Reply:
I'm not sure how to solve that discrepancy either, but note that probably half of the fields fields in this message are arguably envoy-specific, eg, drain type, internal listener config, use_original_dest, api_listener, ignore_global_conn_limit.

Reply:
Maybe we can consider that how much performance impact will be introduced by an extra event loop? 

If the performance impact is low, can we consider to just providing a fixed value for the max number of connections to accept in an event loop? Since the goal is just to prevent the current event loop running for a long time, then other events won't get a chance to be processed.

Or if possible, can we calculate a value based on `/proc/sys/net/core/somaxconn` or `tcp_backlog` (I assume too much connections in an event loop only happened when the backlog size is big also)

Reply:
I think non-API listeners are de facto Envoy specific - our other xDS API users are using API listeners (and maybe something different entirely in the future for something like eBPF). So, I agree with avoiding super gnarly Envoy specifics without asking if they should be separated from generic config in general (e.g. in EDS config for example), but in this specific case I think it's fine. CC @envoyproxy/api-shepherds 

The feature makes sense to me - I agree it's low level, but for tuned resource management this ends up being needed.